[Xcode] Silence "Cache hit|miss" lines in filter-build-webkit

emw-apple opened the pull request:
#### 32a433a5802978ef59eef0540026bd8fc57f5fd0
<pre>
[Xcode] Silence &quot;Cache hit|miss&quot; lines in filter-build-webkit
<a href="https://bugs.webkit.org/show_bug.cgi?id=303892">https://bugs.webkit.org/show_bug.cgi?id=303892</a>
<a href="https://rdar.apple.com/166185115">rdar://166185115</a>

Reviewed by Brandon Stewart and Alexey Proskuryakov.

These appear to print unconditionally after compiler invocations. I
think they are useful for the full log (so we shouldn&apos;t try to disable
them entirely), but not for the filtered output.

* Tools/Scripts/filter-build-webkit:
(shouldIgnoreLine):

Canonical link: <a href="https://commits.webkit.org/304242@main">https://commits.webkit.org/304242@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b490f29f1deaa2d343a37e44f567fab3fc493683

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/134894 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/7315 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/46136 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/142401 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/86765 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/dda17708-24c2-4ece-9218-94dd22290bf5) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/136763 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/7933 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/7162 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/103066 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/70354 "Passed tests") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/473e8752-22ca-45bb-b0ec-699d0b4ac385) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/137840 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/5602 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/120902 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/83912 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/d1b8e994-70e2-4a26-8042-52eab432d9e0) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/5430 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/3044 "Passed tests") | [✅ 🛠 wpe-cairo-libwebrtc](https://ews-build.webkit.org/#/builders/166/builds/2995 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/114647 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/39058 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/145101 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/6990 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/39633 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/111452 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/7046 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/5863 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/111800 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/28390 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/5268 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/117185 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/60916 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/7038 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/35358 "Passed tests") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/6817 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/7054 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/6927 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

aproskuryakov reviewed (approved):
I wonder how much CPU time these separate regexes are using, and whether precompiling a single regex would result in a noticeable improvement.

emw-apple committed:
[Xcode] Silence "Cache hit|miss" lines in filter-build-webkit
https://bugs.webkit.org/show_bug.cgi?id=303892
rdar://166185115

Reviewed by Brandon Stewart and Alexey Proskuryakov.

These appear to print unconditionally after compiler invocations. I
think they are useful for the full log (so we shouldn't try to disable
them entirely), but not for the filtered output.

* Tools/Scripts/filter-build-webkit:
(shouldIgnoreLine):

Canonical link: https://commits.webkit.org/304242@main
